File: sdk/src/main/java/org/coralibre/android/sdk/internal/datatypes/RollingProximityIdentifier.java
package org.coralibre.android.sdk.internal.datatypes;

import java.security.InvalidParameterException;
import java.util.Arrays;

import static org.coralibre.android.sdk.internal.EnFrameworkConstants.RPI_LENGTH;

public class RollingProximityIdentifier {

    private final byte[] data = new byte[RPI_LENGTH];
    private final ENInterval interval;


    public RollingProximityIdentifier(final byte[] rawRPI, final ENInterval interval) {
        if(rawRPI.length != RPI_LENGTH) throw new InvalidParameterException("wrong rawRPI size");
        this.interval = new ENInterval(interval);
        System.arraycopy(rawRPI, 0, data, 0, RPI_LENGTH);
    }

    public byte[] getData() {
        byte[] retVal = new byte[RPI_LENGTH];
        System.arraycopy(data, 0, retVal, 0, RPI_LENGTH);
        return retVal;
    }

    public ENInterval getInterval() {
        return interval;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        RollingProximityIdentifier that = (RollingProximityIdentifier) o;
        return Arrays.equals(data, that.data);
    }

    @Override
    public int hashCode() {
        return Arrays.hashCode(data);
    }
}

File: sdk/src/main/java/org/coralibre/android/sdk/internal/datatypes/RollingProximityIdentifier.kt
package org.coralibre.android.sdk.internal.datatypes

import org.coralibre.android.sdk.internal.EnFrameworkConstants
import java.security.InvalidParameterException

class RollingProximityIdentifier(rawRPI: ByteArray, interval: ENInterval) {
    private val data: ByteArray
    val interval: ENInterval = ENInterval(interval)

    init {
        if (rawRPI.size != EnFrameworkConstants.RPI_LENGTH) throw InvalidParameterException("wrong rawRPI size")
        this.data = rawRPI.copyOf()
    }

    fun getData(): ByteArray {
        val retVal = ByteArray(EnFrameworkConstants.RPI_LENGTH)
        System.arraycopy(data, 0, retVal, 0, EnFrameworkConstants.RPI_LENGTH)
        return retVal
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as RollingProximityIdentifier
        return data.contentEquals(that.data)
    }

    override fun hashCode(): Int {
        return data.contentHashCode()
    }
}
